### ISSUE TITLE
Bug with clearing marking (red background) in selected text

### ISSUE THREAD

**Issue by sasumner** [notepad-plus-plus/notepad-plus-plus#8150]
### Description of the Issue

When a selection is active containing text that has been previously "Mark'd" with a red background, and it is desired to clear the red background ONLY WITHIN THAT SELECTION, the red background is cleared at higher document positions than is covered by the selection (i.e., OUTSIDE the active selection).

### Steps to Reproduce the Issue

1) Create a new tab with the following text:

```
Jack and Jill
Went up the hill
To fetch a pail of water
Jack fell down
And broke his crown,
And Jill came tumbling after.
```

2a) Do a _Mark All_ operation with the following settings:

![image](https://user-images.githubusercontent.com/30118311/79875288-74cbf980-83b7-11ea-91ab-36d0f9b9a797.png)

2b) Observe:

![image](https://user-images.githubusercontent.com/30118311/79875356-8ad9ba00-83b7-11ea-9ad9-53c6fea2e640.png)

3a) Make the following selection of text (selection covers lines 3 and 4) and then do a _Clear all marks_ operation with the following settings:

![image](https://user-images.githubusercontent.com/30118311/79875413-9fb64d80-83b7-11ea-87af-47cc3ec6f01b.png)

3b) Observe **"Actual Behavior"**; too much marking is removed (see line 5 where the bookmark remains but the red background has been eliminated):

![image](https://user-images.githubusercontent.com/30118311/79875462-b2308700-83b7-11ea-9be9-cb9386e7f7cc.png)

3c) Realize that **"Desired Behavior"** is that the red background be removed from only lines 3 and 4, and NOT LINE 5 since line 5 was not in the active selection when _Clear all marks_ was invoked.

### Debug Information

Notepad++ v7.8.4   (32-bit)
Build time : Jan 29 2020 - 01:33:47
Path : C:\...........\npp.7.8.4.bin\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS Name : Windows 10 Enterprise (64-bit)
OS Version : 1809
OS Build : 17763.1039
Plugins : BetterMultiSelection.dll DSpellCheck.dll linefilter2.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll NppMenuSearch.dll NppToolBucket.dll PythonScript.dll qkNppReverseLines.dll XMLTools.dll

**Comment by donho** [comment]
Tested on the following content:
```
Section un.ukrainian
Delete "$INSTDIR\localization\ukrainian.xml"
SectionEnd
Section un.urdu
Delete "$INSTDIR\localization\urdu.xml"
SectionEnd
Section un.uyghur
Delete "$INSTDIR\localization\uyghur.xml"
SectionEnd
Section un.uzbek
Delete "$INSTDIR\localization\uzbek.xml"
SectionEnd
Section un.uzbekCyrillic
Delete "$INSTDIR\localization\uzbekCyrillic.xml"
SectionEnd
Section un.vietnamese
Delete "$INSTDIR\localization\vietnamese.xml"
SectionEnd
```
And I found the following bug:
1. After clicking "Mark All" with the "Section":
![image](https://user-images.githubusercontent.com/90293/80556663-a86cdc00-89d4-11ea-881b-a1849168276b.png)

2. Select 2 lines:
![image](https://user-images.githubusercontent.com/90293/80556765-f681df80-89d4-11ea-80d2-9eddd05ccf8a.png)

3. Check `In Selection` then click `Clear All`, the markers are removed, but not the highlighting:
![image](https://user-images.githubusercontent.com/90293/80556894-64c6a200-89d5-11ea-98c8-1abce39aee28.png)

**Comment by sasumner** [comment]
@donho, YES! That is another way to reproduce the bug.  
I get exactly your results on your data when reproducing the bug in final released version 7.8.6.
